x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/action/TransportValidateTransformAction.java

```diff
@@ -0,0 +1,170 @@
+import org.elasticsearch.xpack.core.transform.action.ValidateTransformAction.Response;
+import org.elasticsearch.xpack.core.transform.transforms.TransformConfig;
+import org.elasticsearch.xpack.transform.transforms.Function;
+import org.elasticsearch.xpack.transform.transforms.FunctionFactory;
+import org.elasticsearch.xpack.transform.utils.SourceDestValidations;
+
+import java.util.Map;
+import java.util.Optional;
+
+public class TransportValidateTransformAction extends HandledTransportAction<Request, Response> {
+
+    private final Client client;
+    private final ClusterService clusterService;
+    private final TransportService transportService;
+    private final boolean isRemoteClusterClientNode;
+    private final SourceDestValidator sourceDestValidator;
+
+    @Inject
+    public TransportValidateTransformAction(
+        TransportService transportService,
+        ActionFilters actionFilters,
+        Client client,
+        IndexNameExpressionResolver indexNameExpressionResolver,
+        ClusterService clusterService,
+        Settings settings,
+        IngestService ingestService
+    ) {
+        this(
+            ValidateTransformAction.NAME,
+            transportService,
+            actionFilters,
+            client,
+            indexNameExpressionResolver,
+            clusterService,
+            settings,
+            ingestService);
+    }
+
+    protected TransportValidateTransformAction(
+        String name,
+        TransportService transportService,
+        ActionFilters actionFilters,
+        Client client,
+        IndexNameExpressionResolver indexNameExpressionResolver,
+        ClusterService clusterService,
+        Settings settings,
+        IngestService ingestService
+    ) {
+        super(name, transportService, actionFilters, Request::new);
+        this.client = client;
+        this.clusterService = clusterService;
+        this.transportService = transportService;
+        this.isRemoteClusterClientNode = DiscoveryNode.isRemoteClusterClient(settings);
+        this.sourceDestValidator = new SourceDestValidator(
+            indexNameExpressionResolver,
+            transportService.getRemoteClusterService(),
+            isRemoteClusterClientNode
+                /* transforms are BASIC so always allowed, no need to check license */
+                ? new RemoteClusterLicenseChecker(client, mode -> true) : null,
+            ingestService,
+            clusterService.getNodeName(),
+            License.OperationMode.BASIC.description()
+        );
+    }
+
+    @Override
+    protected void doExecute(Task task, Request request, ActionListener<Response> listener) {
+        final ClusterState clusterState = clusterService.state();
+        TransformNodes.warnIfNoTransformNodes(clusterState);
+        final DiscoveryNodes nodes = clusterState.nodes();
+
+        if (request.getConfig().getSource().requiresRemoteCluster() && (isRemoteClusterClientNode == false)) {
```

Comment:
as for preview I would _always_ redirect the call if this isn't a transform node.

However, there is a problem: What if there is no transform node? If `defer_validation` is set, I think we should _not_ fail. I suggest to fail in case of no transform node and `defer_validation==false`. If I remember correctly `defer_validation==true` does only simple checks, therefore: 
```
if defer_validation==false:
    if len(transform_nodes) == 0:
        fail("....")
    else:
        redirect(...) # using selectTransformNode to select one with remote client if necessary
else:
    validate(...)
```
This way integrations can still put a transform even if the cluster lacks a transform node.